Auto download data, tree structuring, and phoenix compatibility.

DavoGrant opened the pull request:
This PR expands the code to allow a more seamless and expansive setup:

- Inclusion of phoenix models, closing #6 and #33 .
- Updated the internal workings to use a tree structure. This allowed irregular grids to be more easily queried, and better warnings for poor matches.
- Auto downloading of data.
- Imporved tests.

hrwakeford reviewed (commented):
This looks good!


One suggestion:
In installation.rst it might be worth some explicit statements in the text in bold below 

And now you are ready to go. Stellar models and instrument throughput data
are selected and downloaded automatically**, or custom files input by the user,** at runtime. These data are saved
locally and so are only downloaded once, speeding up subsequent runs. Head
straight to the :doc:`quick start <quick_start>` page to begin computing
limb-darkening coefficients 🚀.

DavoGrant commented:
Great, I have updated the docs and also added a python min version requirement. Looks like we also need at least SciPy==1.8 for the KDtree, and therefore also python>=3.8. I have made this clear on the installation page.

hrwakeford reviewed (approved):
I approve this pull request and release subject to future testing and updates.
